File: android/app/src/main/kotlin/com/greatpeople/card/data/DemoCards.kt
package com.greatpeople.card.data

import android.content.Context
import com.greatpeople.card.data.model.Card
import com.greatpeople.card.data.model.CardTier
import org.json.JSONArray

fun loadDemoCards(context: Context): List<Card> {
    val json = context.assets.open("demo_cards.json").bufferedReader().use { it.readText() }
    val arr = JSONArray(json)
    return (0 until arr.length()).map { i ->
        val obj = arr.getJSONObject(i)
        val key = obj.getString("portraitKey")
        val identitiesArr = obj.getJSONArray("identities")
        Card(
            id = obj.getString("id"),
            figureName = obj.getString("figureName"),
            era = obj.getString("era"),
            gender = obj.getString("gender"),
            identities = (0 until identitiesArr.length()).map { identitiesArr.getString(it) },
            tier = CardTier.valueOf(obj.getString("tier").uppercase()),
            lore = obj.getString("lore"),
            portraitUrl = "file:///android_asset/portraits/portrait_$key.jpeg",
            years = obj.getString("years"),
            characteristics = obj.getString("characteristics"),
            achievement = obj.getString("achievement"),
            politics = obj.getInt("politics"),
            strength = obj.getInt("strength"),
            culture = obj.getInt("culture"),
            wealth = obj.getInt("wealth"),
            intelligence = obj.getInt("intelligence"),
            technique = obj.getInt("technique"),
            belief = obj.getInt("belief"),
            reputation = obj.getInt("reputation"),
        )
    }
}

File: ios/GreatPeople/Models/DemoCards.swift
import Foundation

private struct DemoCardDTO: Decodable {
    let id: String
    let figureName: String
    let era: String
    let gender: String
    let identities: [String]
    let countries: [String]?
    let portraitKey: String
    let lore: String
    let years: String
    let trait: String
    let achievement: String
    let politics: Int
    let strength: Int
    let culture: Int
    let wealth: Int
    let intelligence: Int
    let technique: Int
    let belief: Int
    let reputation: Int

    var asCard: Card {
        Card(
            id: id,
            figureName: figureName,
            era: era,
            gender: gender,
            identities: identities,
            countries: countries ?? [],
            lore: lore,
            portraitUrl: "asset:portrait_\(portraitKey)",
            years: years,
            trait: trait,
            achievement: achievement,
            politics: politics,
            strength: strength,
            culture: culture,
            wealth: wealth,
            intelligence: intelligence,
            technique: technique,
            belief: belief,
            reputation: reputation
        )
    }
}

func loadDemoCards() -> [Card] {
    guard let url = Bundle.main.url(forResource: "demo_cards", withExtension: "json"),
          let data = try? Data(contentsOf: url),
          let dtos = try? JSONDecoder().decode([DemoCardDTO].self, from: data)
    else { return [] }
    return dtos.map(\.asCard)
}
